sycl/include/sycl/accessor.hpp

```diff
@@ -2117,7 +2117,7 @@ class __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   // but for get_pointer() we must return the original pointer.
   // On device, getQualifiedPtr() returns MData, so we need to backjust it.
   // On host, getQualifiedPtr() does not return MData, no need to adjust.
-  PtrType getPointerAdjusted() const {
+  auto getPointerAdjusted() const {
```

Comment:
Why this change was done? I find `PtrType` to be more human-readable than `auto`

Reply:
Problem is that `PtrType` is not decorated so when the underlying pointer is decorated the decoration would be dropped just to be added again by the multi_ptr, which now requires a call to the SPIR-V builtins. Using `auto` makes sure we pick the appropriate decoration too, depending on how `getQualifiedPtr` is defined.